src/main/java/org/zendesk/client/v2/AsyncHttpClientLogger.java

```diff
@@ -12,17 +14,24 @@
 import java.util.List;
 
 import static io.netty.handler.codec.http.HttpHeaders.Names.AUTHORIZATION;
+import static io.netty.handler.codec.http.HttpHeaders.Values.APPLICATION_JSON;
 
 public class AsyncHttpClientLogger
 {
     public static final String BODY_LENGTH = "bodyLength";
     public static final String URL = "url";
     public static final String DIRECTION = "direction";
+    public static final String PROJECT_ID = "projectId";
+
     private final Logger logger;
+    private final ObjectMapper mapper;
+    private final String projectId;
 
-    public AsyncHttpClientLogger(Logger logger)
+    public AsyncHttpClientLogger(Logger logger, ObjectMapper mapper, String projectId)
```

Comment:
maybe just use MDC, if I am not wrong logstash adds params from MDC to log automatically

Reply:
sorry, logback

Reply:
We handle responses in async way, and MDC is per thread